packages/@aws-cdk/aws-lambda-python/lib/bundling.ts

```diff
@@ -81,45 +88,54 @@ export interface BundlingOptions {
 /**
  * Produce bundled Lambda asset code
  */
-export function bundle(options: BundlingOptions): lambda.Code {
-  const { entry, runtime, architecture, outputPathSuffix } = options;
-
-  const stagedir = cdk.FileSystem.mkdtemp('python-bundling-');
-  const hasDeps = stageDependencies(entry, stagedir);
-
-  const depsCommand = chain([
-    hasDeps ? `rsync -r ${BUNDLER_DEPENDENCIES_CACHE}/. ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${outputPathSuffix}` : '',
-    `rsync -r . ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${outputPathSuffix}`,
-  ]);
-
-  // Determine which dockerfile to use. When dependencies are present, we use a
-  // Dockerfile that can create a cacheable layer. We can't use this Dockerfile
-  // if there aren't dependencies or the Dockerfile will complain about missing
-  // sources.
-  const dockerfile = hasDeps
-    ? 'Dockerfile.dependencies'
-    : 'Dockerfile';
-
-  // copy Dockerfile to workdir
-  fs.copyFileSync(path.join(__dirname, dockerfile), path.join(stagedir, dockerfile));
-
-  const image = cdk.DockerImage.fromBuild(stagedir, {
-    buildArgs: {
-      IMAGE: runtime.bundlingImage.image,
-    },
-    platform: architecture.dockerPlatform,
-    file: dockerfile,
-  });
-
-  return lambda.Code.fromAsset(entry, {
-    assetHashType: options.assetHashType,
-    assetHash: options.assetHash,
-    exclude: DEPENDENCY_EXCLUDES,
-    bundling: {
-      image,
-      command: ['bash', '-c', depsCommand],
-    },
-  });
+export class Bundling implements cdk.BundlingOptions {
+  public static bundle(options: BundlingProps): AssetCode {
+    return Code.fromAsset(options.entry, {
+      assetHash: options.assetHash,
+      assetHashType: options.assetHashType,
+      exclude: DEPENDENCY_EXCLUDES,
+      bundling: new Bundling(options),
+    });
+
+  }
+
+  public readonly image: cdk.DockerImage;
+  public readonly command: string[];
+
+
+  constructor(props: BundlingProps) {
+
+    const { entry, runtime, architecture, outputPathSuffix } = props;
+
+    const stagedir = cdk.FileSystem.mkdtemp('python-bundling-');
+    const hasDeps = stageDependencies(entry, stagedir);
+
+    const depsCommand = chain([
+      hasDeps ? `rsync -r ${BUNDLER_DEPENDENCIES_CACHE}/. ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${outputPathSuffix}` : '',
+      `rsync -r . ${cdk.AssetStaging.BUNDLING_OUTPUT_DIR}/${outputPathSuffix}`,
+    ]);
+
+    // Determine which dockerfile to use. When dependencies are present, we use a
+    // Dockerfile that can create a cacheable layer. We can't use this Dockerfile
+    // if there aren't dependencies or the Dockerfile will complain about missing
+    // sources.
+    const dockerfile = hasDeps
+      ? 'Dockerfile.dependencies'
+      : 'Dockerfile';
+
+    // copy Dockerfile to workdir
+    fs.copyFileSync(path.join(__dirname, dockerfile), path.join(stagedir, dockerfile));
+
+    const defaultImage = cdk.DockerImage.fromBuild(stagedir, {
```

Comment:
Along with the comment above about the stage directory, I don't think we need all this file copying anymore. We should be able to just do:

```ts
cdk.DockerImage.fromBuild(path.join(__dirname, '../lib'), {
  ...,
  file: dockerfile,
});

```

The bundling would then use the Dockerfile located within the library, and the app code to be bundled would be mounted into the image at `/asset-input`

Reply:
That makes sense and that appears to be the expected based on what I am reading [here](https://aws.amazon.com/blogs/devops/building-apps-with-aws-cdk/).

However, when I try this, it doesn't mount to `/asset-input`, but maybe to `/var/task`... I looked at the `NodeJsFunction` and `GoFunction` implementation but haven't been able to figure out anything obvious that is missing in here that'd cause it to fail. If you have specifics on what additional properties might be required, I'd appreciate it (it has been tough for me to figure out how bundling is actually occurring).

Reply:
Ah, I think I figured it out. The difference between the other functions and this is that the installation of dependencies in here is happening within the Dockerfile's RUN step (vs. as a command in the others).

The asset-input appears to only available when it is run, not when the Docker image gets built.

I'm happy to modify this to be more inline with the others, but that'll be a slightly bigger change (and might result in slower bundling).

Reply:
Yeah I think the change is worth it. Users will also now be able to supply their own Dockerfile if they want to customize the build and potentially speed up the bundling.

Reply:
Yeah, makes sense. I'll have some more time to finish the rest of this PR over the next week. Thanks! Happy holidays!